### PR TITLE
Camera information

### PR DESCRIPTION
**Pull request by Roccy**
Camera information shown on screen:
* Macro Supported
* Current Focus Mode
* Image Resolution
* Current Estimated Focus distance

Also included a togglebutton to hide the info.

Probably does need an rebase.

### CODE REVIEW

File: FingerBlox/app/src/main/java/org/fingerblox/fingerblox/CameraView.java [hunk 1]
```diff
@@ -5,12 +5,15 @@
...
 import android.hardware.Camera.PictureCallback;
 import android.util.AttributeSet;
+import android.util.Log;
 
 import org.opencv.android.JavaCameraView;
 
 import java.util.ArrayList;
 import java.util.Collections;
 
+import javax.crypto.NullCipher;
```
Comment (jorenham) [inline FingerBlox/app/src/main/java/org/fingerblox/fingerblox/CameraView.java]: Wut?

File: FingerBlox/app/src/main/java/org/fingerblox/fingerblox/CameraView.java [hunk 2]
```diff
@@ -98,4 +101,13 @@ protected void fixFocusToggle() {
         }
         mCamera.setParameters(parameters);
     }
+
+    public Camera.Parameters getCameraParameters() {
+        try {
+            return mCamera.getParameters();
+        } catch (NullPointerException e) {
+            Log.i(TAG, "Could not retrieve camera parameters. Camera in CameraView == NULL");
```
Comment (jorenham) [inline FingerBlox/app/src/main/java/org/fingerblox/fingerblox/CameraView.java:106]: `log.e` would be more  appropriate

File: FingerBlox/app/src/main/java/org/fingerblox/fingerblox/MainActivity.java [hunk 3]
```diff
@@ -117,6 +124,69 @@ public void onClick(View v) {
...
+
+        updateStaticTextViews();
+
+    }
+
+    protected boolean updateStaticTextViews() {
+        Camera.Parameters params = null;
+        try {
+            params = mOpenCvCameraView.getCameraParameters();
+        } catch (NullPointerException e) {
```
Comment (jorenham) [inline FingerBlox/app/src/main/java/org/fingerblox/fingerblox/MainActivity.java]: would be prettier to make this an `if (mOpenCvCameraView == null)`

**Review by jrauhamaa — COMMENTED** [review]
For me, the app crashes with these changes. I'll look into it

Edit: This is because of getSupportedPreviewSizes() in CameraView.java. It's fixed in master branch so never mind.

File: FingerBlox/app/src/main/java/org/fingerblox/fingerblox/CameraView.java (hunk 2 above)
Comment (Roccy) [reply FingerBlox/app/src/main/java/org/fingerblox/fingerblox/CameraView.java:106]: Not true, this happens when the camera is still initializing and when the phone locked. Putting the code in onResume(), onCameraViewStarted() or any of the other functions causes this Exception to be thrown immideately (as the camera is started later than the view).

File: FingerBlox/app/src/main/java/org/fingerblox/fingerblox/MainActivity.java (hunk 3 above)
Comment (Roccy) [reply FingerBlox/app/src/main/java/org/fingerblox/fingerblox/MainActivity.java]: Good one

File: FingerBlox/app/src/main/java/org/fingerblox/fingerblox/CameraView.java (hunk 1 above)
Comment (Roccy) [reply FingerBlox/app/src/main/java/org/fingerblox/fingerblox/CameraView.java]: I.... have no idea xD

File: FingerBlox/app/src/main/java/org/fingerblox/fingerblox/CameraView.java (hunk 2 above)
Comment (jorenham) [reply FingerBlox/app/src/main/java/org/fingerblox/fingerblox/CameraView.java:106]: But but but your throwing the exception anyway below

Comment (Roccy) [reply FingerBlox/app/src/main/java/org/fingerblox/fingerblox/CameraView.java:106]: I had it return null now to use with the try/catch, as both mCamera and getParameters() can be null.

**Review by jorenham — APPROVED** [review]

**Pull request merged**